src/babel-custom.js

```diff
@@ -0,0 +1,160 @@
+import { createConfigItem } from '@babel/core';
+import babelPlugin from 'rollup-plugin-babel';
+import merge from 'lodash.merge';
+import { isTruthy } from './utils';
+
+const ESMODULES_TARGET = {
+	esmodules: true,
+};
+
+const mergeConfigItems = (type, ...configItemsToMerge) => {
+	const mergedItems = [];
+
+	configItemsToMerge.forEach(configItemToMerge => {
+		configItemToMerge.forEach(item => {
+			const itemToMergeWithIndex = mergedItems.findIndex(
+				mergedItem => mergedItem.file.resolved === item.file.resolved,
+			);
+
+			if (itemToMergeWithIndex === -1) {
+				mergedItems.push(item);
+				return;
+			}
+
+			mergedItems[itemToMergeWithIndex] = createConfigItem(
+				[
+					mergedItems[itemToMergeWithIndex].file.resolved,
+					merge(mergedItems[itemToMergeWithIndex].options, item.options),
+				],
+				{
+					type,
+				},
+			);
+		});
+	});
+
+	return mergedItems;
+};
+
+const createConfigItems = (type, items) => {
+	return items.map(({ name, ...options }) => {
+		return createConfigItem([require.resolve(name), options], { type });
+	});
+};
+
+export const config = (config, { customOptions }) => {
```

Comment:
I moved and changed this file, so GitHub won't diff this correctly. I lifted `config` to the module scope and made it exportable.